pkg/compact/downsample/downsample.go

```diff
@@ -450,7 +451,7 @@ func expandChunkIterator(it chunkenc.Iterator, buf *[]sample) error {
 	// If it does, we skip it.
 	lastT := int64(0)
 
-	for it.Next() {
+	for it.Next() != chunkenc.ValNone {
```

Comment:
`Next()` returns the value type at the next position: https://github.com/prometheus/prometheus/blob/965274259f993801d63f553639d248588287120b/tsdb/chunkenc/chunk.go#L101
`ValNone` is returned when the iterator is exhausted. While it could also return other values than `ValFloat` for now (in this PR) we don't ingest them and it can be ignored. So checking for `chunkenc.ValNone` to end iterating should be enough.